File: backend/src/main/java/nc/kibagami_nc/vencosenc/service/BidService.java
package nc.kibagami_nc.vencosenc.service;

import java.util.List;

import nc.kibagami_nc.vencosenc.dto.BidDto;

public interface BidService {

    // Renvoie toutes les annonces
    List<BidDto> findAll();

    // Renvoie une annonce par son id
    BidDto findById(Long id);

    // Cree une nouvelle annonce
    BidDto create(BidDto dto);

    // Modifie une annonce existante
    BidDto update(Long id, BidDto dto);

    // Supprime une annonce par son id
    void delete(Long id);
}

File: app/src/main/frontend/src/app/shared/services/bid.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { Bid } from '../models/bid.model';

// Service qui parle au backend Spring (endpoint /api/bid)
@Injectable({ providedIn: 'root' })
export class BidService {

  private http = inject(HttpClient);
  private apiUrl = 'http://localhost:8080/api/bid';

  // Recupere toutes les annonces
  findAll(): Observable<Bid[]> {
    return this.http.get<Bid[]>(this.apiUrl);
  }

  // Recupere une annonce par son id
  findById(id: number): Observable<Bid> {
    return this.http.get<Bid>(`${this.apiUrl}/${id}`);
  }
}
